codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/TraitInfo.kt

```diff
@@ -20,7 +22,27 @@ data class TraitInfo(
     val constraintViolationVariant: Writable,
     val asValidationExceptionField: Writable,
     val validationFunctionDefinition: (constraintViolation: Symbol, unconstrainedTypeName: String) -> Writable,
-)
+    private val testCases: List<Writable> = listOf(),
+) {
+    companion object {
+        fun testCases(constraintsInfo: List<TraitInfo>): List<Writable> {
+            val annotatedTestCases = constraintsInfo.flatMap { it.testCases }
+                .map { testCase ->
+                    writable {
+                        rustTemplate(
```

Comment:
We could use `RustWriter.unitTest` here, and the list of test writables provided by `TraitInfo` would then just comprise the test function bodies.

Reply:
Done in 680857323d283270d8342f2f5b80d3c872fcd137.